platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/MerkleDb.java

```diff
@@ -163,26 +161,29 @@ public final class MerkleDb {
      * created first. If the path is {@code null}, the default MerkleDb path is used instead.
      *
      * @param path Database storage dir. If {@code null}, the default MerkleDb path is used
+     * @param configuration platform configuration
      * @return Virtual database instance that stores its data in the specified path
      */
-    public static MerkleDb getInstance(final Path path) {
-        return getInstance(path, ConfigurationHolder.getConfigData(MerkleDbConfig.class));
-    }
+    public static MerkleDb getInstance(final Path path, final @NonNull Configuration configuration) {
+        requireNonNull(configuration);
 
-    static MerkleDb getInstance(final Path path, final MerkleDbConfig config) {
-        return instances.computeIfAbsent(path != null ? path : getDefaultPath(), p -> new MerkleDb(p, config));
+        return instances.computeIfAbsent(
+                path != null ? path : getDefaultPath(configuration), p -> new MerkleDb(p, configuration));
     }
 
     /**
      * A database path (storage dir) to use for new or restored data sources
      *
+     * @param configuration platform configuration
      * @return Default instance path
      */
-    private static Path getDefaultPath() {
+    private static Path getDefaultPath(final @NonNull Configuration configuration) {
```

Comment:
This is another example when a static field (default instance path) is initialized from a non-static context (configuration).

The right way to solve it would be to move MerkleDb initialization to PlatformContext. Multiple MerkleDb objects should be possible in a single JVM, each with its own configuration and list of instances. I suggest to file a follow-up ticket for this

Reply:
https://github.com/hashgraph/hedera-services/issues/15945